lib/Sapi.php

```diff
@@ -89,9 +89,28 @@ public static function sendResponse(ResponseInterface $response)
         if (null !== $contentLength) {
             $output = fopen('php://output', 'wb');
             if (is_resource($body) && 'stream' == get_resource_type($body)) {
-                if (PHP_INT_SIZE !== 4) {
+                if (PHP_INT_SIZE > 4) {
                     // use the dedicated function on 64 Bit systems
-                    stream_copy_to_stream($body, $output, (int) $contentLength);
+                    // a workaround to make PHP more possible to use mmap based copy, see https://github.com/sabre-io/http/pull/119
+                    $left = (int) $contentLength;
+                    // copy with 4MiB chunks
+                    $chunk_size = 4 * 1024 * 1024;
+                    stream_set_chunk_size($output, $chunk_size);
+                    // If this is a partial response, flush the beginning bytes until the first position that is a multiple of the page size.
+                    $contentRange = $response->getHeader('Content-Range');
+                    // Matching "Content-Range: bytes 1234-5678/7890"
+                    if (null !== $contentRange && preg_match('/^bytes\s([0-9]*)-([0-9]*)\//i', $contentRange, $matches) && '' !== $matches[1]) {
```

Comment:
Improve regex by https://github.com/sabre-io/http/pull/136

Reply:
IIRC if you want everything except the first 4 bytes, sending `Content-Range: bytes 5-`  is legal, but this regex won't validate it

Reply:
The `Content-Range` header is in the response of the server, not the request from user:
See https://tools.ietf.org/html/rfc7233#section-4.2 and https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Range

Reply:
I also made a comment on that PR.